HeterogeneousCore/CUDACore/python/ProcessAcceleratorCUDA.py

```diff
@@ -4,27 +4,40 @@
 
 class ProcessAcceleratorCUDA(cms.ProcessAccelerator):
     def __init__(self):
-        super(ProcessAcceleratorCUDA,self).__init__()
+        super(ProcessAcceleratorCUDA, self).__init__()
         self._label = "gpu-nvidia"
+        self._required_tool = "cuda-gcc-support"
+        self._runtime_check = "cudaIsEnabled"
+        self._available = (os.system(f"scram tool tag {self._required_tool} TOOLNAME >& /dev/null") == 0)
+        self._enabled = self._available and (os.system(self._runtime_check) == 0)
```

Comment:
The calls to `os.system()` must be done in `enabledLabels()` and `apply()`. (In the case of offline processing) The `ProcessAccelerator` constructors are run on the machine where the configuration is pickled (including the member variables here), instead of the worker node whose state should dictate he behavior.

I'll clarify the documentation of ProcessAccelerator of this aspect.

Reply:
TBH I did not read any documentation :-/